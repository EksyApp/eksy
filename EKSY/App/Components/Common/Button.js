import React, {Component} from 'react';
import { Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types'
import * as Theme from '../../Theme'

export class Button extends Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress} style={styles.buttonStyle}>
        <Text style={styles.textStyle}>
          {this.props.children}
        </Text>
      </TouchableOpacity>
    )
  }
}

Button.propTypes = {
  onPress: PropTypes.func,
  children: PropTypes.string
}


const styles = {
  textStyle: {
    alignSelf: 'center',
    color: Theme.accentColor,
    fontSize: 16,
    fontWeight: '600',
    paddingTop: 10,
    paddingBottom: 10
  },
  buttonStyle: {
    height: 40,
    alignSelf: 'stretch',
    backgroundColor: Theme.buttonColor,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: Theme.accentColor,
    marginLeft: 5,
    marginRight: 5,
    marginTop: 5
  }
};

// export default Button;
