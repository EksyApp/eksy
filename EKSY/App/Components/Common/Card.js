import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';
import {frontgroundColor} from '../../Theme'

const Card = (props) => (
  <View {...props} style={[styles.container, props.style]}>
  </View>
);

export default Card;

const styles = StyleSheet.create({
  container: {
	  backgroundColor: frontgroundColor,
    borderWidth:1,
    borderRadius: 2,
    borderColor:'#ddd',
    borderBottomWidth: 0,
    shadowColor: '#000',
	  shadowOffset: {width: 0, height: 2},
	  shadowOpacity: 0.1,
	  shadowRadius: 2,
    elevation: 1,
    marginLeft: 5,
    marginTop: 10,
  },
});
