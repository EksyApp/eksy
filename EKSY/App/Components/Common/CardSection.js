import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';

const CardSection = (props) => (
    <View {...props} style={[styles.container, props.style]}>
    </View>
);

export { CardSection };

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    padding: 5,
    backgroundColor: '#fff',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    borderColor: '#ddd',
    position: 'relative',
  },
});
