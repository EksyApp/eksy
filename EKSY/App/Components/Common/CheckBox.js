import React, {Component} from 'react'
import {StyleSheet} from 'react-native'
import * as Theme from '../../Theme'
const BasicCheckBox = require('react-native-elements').CheckBox
import PropTypes from 'prop-types';
import {StyleShape} from "../../Utils/PropTypeShapes";

export class CheckBox extends Component {

	constructor(props) {
		super(props)

		this.state = {
			checked: this.props.checked != null ? this.props.checked : false
		}

	}

	componentWillReceiveProps(props) {
		if(this.props.name !== props.name) {
			this.setState({checked: props.checked !== null ? props.checked : this.state.checked})
		}
	}

	_handlePress() {
		this.setState({checked: !this.state.checked})
		this.props.onPress(this.props.name, !this.state.checked);
	}

	render() {
		return (
				<BasicCheckBox
						{...this.props}
						containerStyle={[style.checkboxContainer, this.props.containerStyle]}
						textStyle={[style.checkboxText, this.props.textStyle]}
						checked={this.state.checked}
						onPress={() => {this._handlePress()}}
				/>
		)

	}

}

const style = StyleSheet.create({
	checkboxContainer: {
		backgroundColor: Theme.frontgroundColor
	},

	checkboxText: {
		color: Theme.fontColor
	}
})

CheckBox.propTypes = {
	checked: PropTypes.bool,
	name: PropTypes.string,
	containerStyle: StyleShape,
	textStyle: StyleShape
}