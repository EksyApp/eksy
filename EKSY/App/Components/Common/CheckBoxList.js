import React, {Component} from 'react'
import {CheckBox} from './index'
import {ListView} from 'react-native'
import PropTypes from 'prop-types';
import {FiltersShape, StyleShape} from '../../Utils/PropTypeShapes'

export default class CheckBoxList extends Component {
	
	constructor(props) {
		super(props)
		
		const datasource = new ListView.DataSource({rowHasChanged: (r1,r2) => r1 !== r2})
		
		this.state = {
			dataSource: datasource.cloneWithRows(this.props.data)
		}
		
	}
	
	componentWillReceiveProps(props) {
		if(props.data !== this.props.data) {
			const dataSource = this.state.dataSource.cloneWithRows(props.data || []);
			this.setState({dataSource})
		}
	}
	
	render() {
		return (
				<ListView
						style = {this.props.listStyle}
						dataSource={this.state.dataSource}
						renderRow = {(checkbox) => <CheckBox
								checked={checkbox.checked}
								name={checkbox.name}
								title={this.props.titleKey ? checkbox[this.props.titleKey] : checkbox.stateDescription}
								onPress={(name, checked) => {this.props.onPress(name,checked)}}
						/>}
						enableEmptySections={true}
				/>
		)
	}
	
}

CheckBoxList.propTypes = {
	data: FiltersShape,
	listStyle: StyleShape,
	titleKey: PropTypes.string,
	onPress: PropTypes.func,
}