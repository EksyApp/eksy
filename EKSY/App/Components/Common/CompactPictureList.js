import React, {Component} from 'react'
import {ListView, View, StyleSheet} from 'react-native'
import Image from "./Image";

const imagesPerRow = 4

export default class CompactPictureList extends Component {
	
	constructor(props) {
		super(props)
		
		
		
		const datasource = new ListView.DataSource({rowHasChanged: (r1,r2) => r1 !== r2})
		
		this.state = {
			dataSource: datasource.cloneWithRows(this._makeGroups(this.props.data, imagesPerRow))
		}
		
	}
	
	componentWillReceiveProps(props) {
		if(props.data !== this.props.data) {
			const dataSource = this.state.dataSource.cloneWithRows(this._makeGroups(props.data, imagesPerRow) || []);
			this.setState({dataSource})
		}
	}
	
	_makeGroups(data, itemsPerGroup) {
		let groups = []
		for (var i = 0; i < data.length; i += itemsPerGroup) {
			let group = []
			for (var j = i; j < Math.min(data.length, i+itemsPerGroup); j++) {
				group.push(data[j])
			}
			groups.push(group)
		}
		return groups
	}
	
	renderImageRow(data) {
		return(
				<View style={styles.rowContainer}>
					{data.map((datum, index) => {
						return(
								<View style={styles.imageContainer} key={index}>
									<Image source={{uri: datum.uri}} style={styles.image} />
								</View>
						)
					})}
				</View>
		)
	}
	
	render() {
		return(
				<ListView
						style = {this.props.listStyle}
						dataSource={this.state.dataSource}
						renderRow = {(data) => this.renderImageRow(data)}
				/>
		)
	}
	
}

const styles = StyleSheet.create({
	rowContainer: {
		height: 100,
		flexDirection: 'row'
	},
	
	imageContainer: {
		width: 100/imagesPerRow + "%",
		justifyContent: 'center',
		height: '100%',
	},
	
	image: {
		width: '100%',
		height: '100%',
		resizeMode: 'cover'
	}
})