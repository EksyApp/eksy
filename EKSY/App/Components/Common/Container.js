import React, {Component} from 'react'
import * as Theme from '../../Theme'
import {ScrollView, View, StyleSheet} from 'react-native'
import Header from './Header'
import PropTypes from 'prop-types'

export default class Container extends Component {
	
	render() {
		return(
				<View style={styles.container}>
					<Header backButton={this.props.backButton} title={this.props.title} menuButtonPress={this.props.menuButtonPress}/>
					<ScrollView {...this.props} style={[styles.container, this.props.style]}/>
				</View>
		)
	}
	
}

Container.propTypes = {
		...ScrollView.propTypes,
	backButton: PropTypes.bool,
	title: PropTypes.string,
	menuButtonPress: PropTypes.func
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor
	},
	
})