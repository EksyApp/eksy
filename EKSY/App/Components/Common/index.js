export * from './Button'
export * from './Card'
export * from './CardSection'
export * from './Divider'
export * from './Header'
export * from './Label'
export * from './Input'
export * from './TextInputArea'
export * from './TextArea'
