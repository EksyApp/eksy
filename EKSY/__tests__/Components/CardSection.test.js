import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {CardSection} from "../../App/Components/Common/CardSection";

test('CardSection renders correctly', () => {
	const tree = renderer.create(
			<CardSection />
	).toJSON();
	expect(tree).toMatchSnapshot();
});