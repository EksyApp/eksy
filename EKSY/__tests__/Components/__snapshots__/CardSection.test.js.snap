// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSection renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgb(112,112,112)",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      },
      undefined,
    ]
  }
/>
`;
